File: app/src/main/java/br/com/vicentec12/mygames/ui/splash/di/SplashComponent.java
package br.com.vicentec12.mygames.ui.splash.di;

import br.com.vicentec12.mygames.ui.splash.SplashActivity;
import dagger.Subcomponent;

@Subcomponent
public interface SplashComponent {

    @Subcomponent.Factory
    interface Factory {

        SplashComponent create();

    }

    void inject(SplashActivity mSplashActivity);

}

File: app/src/main/java/br/com/vicentec12/mygames/ui/splash/di/SplashComponent.kt
package br.com.vicentec12.mygames.ui.splash.di

import br.com.vicentec12.mygames.di.ActivityScope
import br.com.vicentec12.mygames.ui.splash.SplashActivity
import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = [SplashModule::class])
interface SplashComponent {

    @Subcomponent.Factory
    interface Factory {

        fun create(): SplashComponent

    }

    fun inject(mSplashActivity: SplashActivity)

}